### ISSUE TITLE
wrong POSIX Regex for directory and file matching

### ISSUE THREAD

**Issue by iptechnology-coelli** [svlsResearch/ha-mikrotik#8]
in ha_pushbackup.script you use `[/file find name~"^[^H][^A][^_]" and type="directory"]` and ` [find name~\"^[^H][^A][^_]\" and type!=\"directory\"]`
This REGEX will skip ALL folders/files starting with an "H", all folders/files with an "A" as second char and all folders/files with a "`_`" as a third char.
I think that the right REGEX should be `^([^H]|H($|[^A]|A($|[^_]))).*` and in MK:
`[/file find name~"^([^H]|H(\$|[^A]|A(\$|[^_]))).*" and type="directory"]`
This will skip all folders/files starting with "`HA_`"

Thanks
Alessandro

**Comment by nathanfaber** [comment]
Good catch. I think I am actually going to get rid of the regex entirely and switch to something like:
````
:foreach k in [/file find type="directory"] do={
   :local xferfile [/file get $k name]
   if ([:pick "$xferfile" 0 3] != "HA_") do={ :put "Create dir: $xferfile" }
}
````

**Comment by nathanfaber** [comment]
Implemented 0ae761c8e4860b72b4bada9f709ecfef30dfe833
